app/code/Magento/ConfigurableProduct/Model/Inventory/ParentItemProcessor.php

```diff
@@ -21,27 +22,11 @@
 class ParentItemProcessor implements ParentItemProcessorInterface
 {
     /**
-     * @var ChangeParentStockStatus
-     */
-    private $changeParentStockStatus;
-
-    /**
-     * @param Configurable $configurableType
-     * @param StockItemCriteriaInterfaceFactory $criteriaInterfaceFactory
-     * @param StockItemRepositoryInterface $stockItemRepository
-     * @param StockConfigurationInterface $stockConfiguration
-     * @param ChangeParentStockStatus|null $changeParentStockStatus
-     * @SuppressWarnings(PHPMD.UnusedFormalParameter) Deprecated dependencies
+     * @param ChangeParentStockStatus $changeParentStockStatus
      */
     public function __construct(
-        Configurable $configurableType,
-        StockItemCriteriaInterfaceFactory $criteriaInterfaceFactory,
-        StockItemRepositoryInterface $stockItemRepository,
-        StockConfigurationInterface $stockConfiguration,
-        ?ChangeParentStockStatus $changeParentStockStatus = null
+        private readonly ChangeParentStockStatus $changeParentStockStatus
     ) {
-        $this->changeParentStockStatus = $changeParentStockStatus
-            ?? ObjectManager::getInstance()->get(ChangeParentStockStatus::class);
```

Comment:
I think this part should be reverted for upstream/backwards compatibility. It looks functionally equivalent. Upstream will update the parameters eventually.